bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -179,6 +209,29 @@ private void refresh() {
         }
     }
 
+    private Unit<?> getPriceUnit() {
+        Unit<?> priceUnit = UnitUtils.parseUnit("EUR/kWh");
+        if (priceUnit == null) {
+            priceUnit = CurrencyUnits.BASE_ENERGY_PRICE;
+            logger.info("Using {} instead of EUR/kWh, because it is not available", priceUnit);
+        }
+        return priceUnit;
+    }
+
+    private void createAndSendTimeSeries(String channelId, Function<AwattarPrice, Double> valueFunction) {
+        SortedSet<AwattarPrice> prices = getPrices();
+        Unit<?> priceUnit = getPriceUnit();
+        if (prices == null) {
+            return;
+        }
+        TimeSeries timeSeries = new TimeSeries(TimeSeries.Policy.REPLACE);
+        prices.forEach(p -> {
+            timeSeries.add(Instant.ofEpochMilli(p.timerange().start()),
+                    new QuantityType<>(valueFunction.apply(p) / 100.0, priceUnit));
```

Comment:
Maybe comment this calculation to €, it took me a while to figure this.

Reply:
Which part? Looks pretty clear to me, so I don't know what needs explanation here.

Reply:
The division by 100 to go to € instead of €ct. But it's not that important.